.github/workflows/molecule.yaml

```diff
@@ -58,7 +58,8 @@ jobs:
     - name: Check whether the role is modified
       id: check_role
       run: |
-        if [[ $(git diff --name-only roles | grep "${{ matrix.tested_role }}/") ]]; then
+        git fetch origin main
+        if [[ $(git diff --name-only origin/main roles | grep "${{ matrix.tested_role }}/") ]]; then
```

Comment:
you probably want to use the target branch for the PR here, not main. Could probably use $GITHUB_BASE_REF